### PR TITLE
form macros: base most of these on the upstream toolkit's copy

### PR DESCRIPTION
**Pull request by risicle**
Story https://trello.com/c/W3d7VMrO/72-supplier-app-use-frontend-toolkit-macros-3

Depends on https://github.com/alphagov/digitalmarketplace-frontend-toolkit/pull/375

Don't know how much more I need to say.

### CODE REVIEW

File: app/templates/macros/toolkit_forms.html [hunk 1]
```diff
@@ -1,300 +1,102 @@
...
-    optional=question_content.optional,
-    id=question_content.id,
+  {{ upstream_forms.pricing(question_content, service_data, errors,
     error=errors.get(question_content.id)['message']|question_references(kwargs.get_question),
-    question_number=kwargs.question_number,
-    message=kwargs.message
-  %}
-    {% include "toolkit/forms/pricing.html" %}
-  {% endwith %}
+    hint=(question_content.hint or '')|question_references(kwargs.get_question),
```
Comment (TheDoubleK) [inline app/templates/macros/toolkit_forms.html]: `hint` and `question` didn't previously have the `|question_references` - why have you added it for this pricing macro, but not for multiquestion and boolean_list?  Do we know we need it here, or it just feels like we might?

**Comment by TheDoubleK** [comment]
This looks good as far as I can see.  I think I'd like to give it a quick try-out once the toolkit version is merged, before thumbsing up.

File: app/templates/macros/toolkit_forms.html (hunk 1 above)
Comment (risicle) [reply app/templates/macros/toolkit_forms.html]: Ah yes I did add that to some of the parameters for symmetry, assuming they were omissions. I can't think how including this feature generally across all similar fields could backfire too much, and it's the "least surprising" thing for the macros to do for a macro user.

**Comment by risicle** [comment]
Please do - the quickest way of doing so is modifying `bower.json` to point to `https://github.com/alphagov/digitalmarketplace-frontend-toolkit.git#ris-form-text-field-add-unit-vars`

**Comment by risicle** [comment]
Interesting that there were no other merges into this repo in the last 12 days so I didn't need to rebase.

**Pull request merged**